excretor/src/routing.rs

```diff
@@ -73,15 +81,57 @@ mod handlers {
         per_page: u32,
     }
 
-    // basic handler that responds with a static string
+    #[derive(Deserialize)]
+    pub struct AuthCallback {
+        code: String,
+    }
+
+    #[derive(Deserialize)]
+    pub struct AuthToken {
+        token: Option<String>,
+    }
+
+    const FORBIDDEN_MSG: &str = "Mortals are forbidden from accessing the site";
+
     pub(super) async fn root(
         State(state): State<RouterState>,
+        Query(auth_token): Query<AuthToken>,
     ) -> Result<(StatusCode, Response), AppError> {
+        let ask_auth = false; // set to true to prompt for authentication
```

Comment:
Can this be made an env var?

Reply:
yeah, review pls [2794c12](https://github.com/kossiitkgp/top-secret-stuff/pull/45/commits/2794c123f88c548e2bcd65f8441cd5e0e0aa301e)